spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/dsl/TaskDefinitionException.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core.dsl;
+
+/**
+ * Root exception for DSL parsing related exceptions. Rather than holding a
+ * hard coded string indicating the problem, it records a message key and
+ * the inserts for the message. See {@link DSLMessage} for the list of all
+ * possible messages that can occur.
+ *
+ * @author Andy Clement
+ */
+@SuppressWarnings("serial")
+public class TaskDefinitionException extends RuntimeException {
```

Comment:
is this a direct copy of StreamDefinitionException? if so, let's create a ParsingException (my comment above wrt the checkpointed subclass would also be addressed)